c++/src/kj/compat/tls-win32.c++

```diff
@@ -0,0 +1,67 @@
+
+#ifdef _WIN32
+#include <openssl/x509.h>
+
+#include <kj/win32-api-version.h>
+#include <windows.h>
+#include <wincrypt.h>
+#undef CONST
+#include <kj/windows-sanity.h>
+
+#include <kj/debug.h>
+#include <kj/vector.h>
+
+kj::Exception getOpensslError() {
+  // Call when an OpenSSL function returns an error code to convert that into an exception.
+
+  kj::Vector<kj::String> lines;
+  while (unsigned long long error = ERR_get_error()) {
+#ifdef SSL_R_UNEXPECTED_EOF_WHILE_READING
+    // OpenSSL 3.0+ reports unexpected disconnects this way.
+    if (ERR_GET_REASON(error) == SSL_R_UNEXPECTED_EOF_WHILE_READING) {
+      return KJ_EXCEPTION(DISCONNECTED,
+          "peer disconnected without gracefully ending TLS session");
+    }
+#endif
+
+    char message[1024]{};
+    ERR_error_string_n(error, message, sizeof(message));
+    lines.add(kj::heapString(message));
+  }
+  kj::String message = kj::strArray(lines, "\n");
+  return KJ_EXCEPTION(FAILED, "OpenSSL error", message);
+}
+
+KJ_NORETURN(void throwOpensslError());
+void throwOpensslError() {
+  // Call when an OpenSSL function returns an error code to convert that into an exception and
+  // throw it.
+
+  kj::throwFatalException(getOpensslError());
+}
+
+// Adds windows CA store certificates to OpenSSL's ca store
+// Taken from https://stackoverflow.com/questions/9507184/can-openssl-on-windows-use-the-system-certificate-store
+void updateOpenSSLCAStoreWithWindowsCertificates(void* ctx) {
```

Comment:
I guess this function is the only new code in this file -- the other functions above are copy/pasted from `tls.c++`.

Can we just put this in `tls.c++`? Having a whole separate file for this seems unnecessary.

Reply:
I will deduplicate soon, it will be cleaner.
I tried to put it in tls.c++ and been fighting with the compiler for a day but there seems to be something from the windows headers that's ruining compilation and I can't figure out why
This is the error I'm getting
![image](https://github.com/user-attachments/assets/87682fda-a73f-4040-a684-44b859cf0543)

Reply:
It looks like `wincrypt.h` defines `X509_NAME` as a macro constant. You can probably just `#undef X509_NAME` after including `wincrypt.h`.

https://stackoverflow.com/questions/49504648/x509-name-macro-in-c-wont-compile

Reply:
Thanks! that's really helpful sad I couldn't figure that out